index.html

```diff
@@ -89,44 +143,166 @@
 <script>
   var data = [];
   var options = {
-    yearRange: 'all'
+    categoryRange: 'all'
   };
+  
+
+  var svg, x, y;
 
-  var svg;
+  var margin = { top: 15, right: 15, bottom: 30, left: 45 } ; //spacing b/w svg and g
+
+  var width = 840 - margin.right - margin.left;
+  var height = 500 - margin.top - margin.bottom;
 
   // FETCH DATA
 
+  d3.json('school_data.json', function(error, json) {
+    if (error) { throw json;} //if error is defined, then don't do anything
+
+
+    
+
+    data = json;
+
+    setup(); //must occur after fetching the data
+    update();
+  });
+
   function setup() {
     // SVG
+   
 
     svg = d3.select('#chart')
-      .append('svg');
+      .append('svg')
+      .attr('width', width + margin.right + margin.left)
+      .attr('height', height + margin.top + margin.bottom)
+      .append('g')
+      .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
 
+
+ 
     // SCALES
 
+    var padding = 20;
+
+    x = d3.scale.linear()
+      .domain([0, d3.max(data, function (d) { return d.Reading_proficient/100; })]) //finds highest value of reading scores
+      .range([padding, width-margin.right])
```

Comment:
We can set the range to `[0, width]`. On line 180, we inset the group that we're writing everything to by the top and left margins, obviating the need for starting at `padding`. And we've already subtracted the left and right margins from width when we first declared it on line 154, adding them back in when we size the SVG on line 177.